### DISCUSSION TITLE
Cannot use :global to scope Tailwind?

### DISCUSSION THREAD

**Discussion by MartinDoyleShopify** [tailwindlabs/tailwindcss#8412] [Help]
I have the following global css file that's loaded in from my main application framework file:

```css
:global {
  @tailwind base;
  @tailwind components;
  @tailwind utilities;
}
```

Unfortunately, this triggers the rule in [detectNesting.js](https://github.com/tailwindlabs/tailwindcss/blob/master/src/lib/detectNesting.js) which outputs the message _"Nested @tailwind rules were detected"_.  If I remove the `:global` scoping then the CSS modules within the app cannot see the Tailwind classes. If I leave it in, then my production build fails because it forces failure on warnings (and even using `ignoreWarnings` in my webpack config doesn't seem to prevent this).

Is there some alternative strategy for doing this? Or alternatively, is it actually considered "ok" to do this and in fact there's a missing exception in this rule-checker?

**Comment by MartinDoyleShopify** [comment]
Have created a PR (https://github.com/tailwindlabs/tailwindcss/pull/8418) to allow this type of special-case nesting. Would love some comments and even approval on it, if it's appropriate?

**Reply by thecrypticace** [reply]
Can you provide some more details about your setup? I just set up a small webpack project with CSS modules and it appears that I am not required to use `:global` but I admittedly don't use them a lot.

**Reply by MartinDoyleShopify** [reply]
I'm coming onto the project fairly recently, so I'm not super familiar with the setup, although I do understand what's going on, I think. Let me put something together for you.

**Reply by thecrypticace** [reply]
@MartinDoyleShopify did you get a chance to do this at all? Right now the PR, as written, wouldn't work for variants unless you used an explicit `@tailwind variants` rule.

**Reply by MartinDoyleShopify** [reply]
Unfortunately I've not had the time to do this with other work commitments. That's interesting to know though. Thank you. If you need to clean up, I'm happy to close this discussion/PR and pick it up again another time when I get time to address it properly.

**Comment by tomprats** [comment]
@MartinDoyleShopify I did this previously and it worked fine, but on a new project with the latest versions of things I am running into the warning you mentioned above. I haven't been able to work around it yet.

@thecrypticace The use-case I have is that I'm using CSS modules to locally scope all the css in my app by default. And then I manually exclude custom helpers (and Tailwind). I do this since I use tailwind classes for most things but want a component to be able to have some custom css and not worry about a conflict.

Below is an example config, where by specifying `modules` as an object, it enables css modules and transforms all class names to use the `localIdentName`. To avoid that occuring for Tailwind classes, I used to wrap their initial import/directive in `:global` as seen in the post above.

Is the above PR still a possibility or could I make a new one that would allow `:global` around `@tailwind`?

```
rules: [
  ...
  {
    test: /\.(scss|css)$/,
    use: [
      {
        loader: "css-loader",
        options: {
          modules: {
            localIdentName: "[local]-[hash:base64:5]"
          },
          sourceMap: true,
          importLoaders: 2
        }
      },
      {
        loader: "postcss-loader",
        options: {
          sourceMap: true
        }
      },
      {
        loader: "sass-loader",
        options: {
          sourceMap: true
        }
      }
    ]
  }
]
```

**Reply by tomprats** [reply]
@thecrypticace Here is the discussion that connects to https://github.com/tailwindlabs/tailwindcss/pull/12635
